### PR TITLE
Fix controller detection for Pico Neo3 Link

### PR DESCRIPTION
**Pull request by HoLo85**
Fixes the detection of the controllers for the Pico Neo3 Link headset. With the introduction of the Pico 4 the HMD was detected as this model which broke the detection of the controller.

### CODE REVIEW

File: CHANGELOG.md
Comment (Vixea) [inline CHANGELOG.md:1]: No need to change the change log that will be done by zarik

File: alvr/client_core/src/connection.rs [hunk 1]
```diff
@@ -73,8 +73,9 @@ pub struct ConnectionContext {
 
 fn set_hud_message(event_queue: &Mutex<VecDeque<ClientCoreEvent>>, message: &str) {
     let message = format!(
-        "ALVR v{}\nhostname: {}\nIP: {}\n\n{message}",
+        "ALVR v{}\nHMD {}\nhostname: {}\nIP: {}\n\n{message}",
```
Comment (Vixea) [inline alvr/client_core/src/connection.rs:76]: Could you explain this change seems unnecessary and should be done in a different pr

**Review by Vixea — CHANGES_REQUESTED** [review]

File: CHANGELOG.md
Comment (HoLo85) [reply CHANGELOG.md:1]: Thank you, will revert this change.

File: alvr/client_core/src/connection.rs (hunk 1 above)
Comment (HoLo85) [reply alvr/client_core/src/connection.rs:76]: Needed this to check if the model of the HMD was detected correct directly in the app before connecting it to the server. Since I didn't see the model in the log, I thought displaying the name directly in the app would be easier and faster instead of always connecting the client to the server or searching the log

File: alvr/client_core/src/platform/mod.rs [hunk 2]
```diff
@@ -69,6 +71,7 @@ pub fn platform() -> Platform {
             ("Oculus", _, "seacliff") => Platform::QuestPro,
             ("Oculus", _, _) => Platform::QuestUnknown,
             ("Pico", "Pico Neo 3", _) => Platform::PicoNeo3,
+            ("Pico", "Pico Neo3 Link", _) => Platform::PicoNeo3Link,
             ("Pico", _, _) => Platform::Pico4,
```
Comment (HoLo85) [inline alvr/client_core/src/platform/mod.rs:75]: Question to Neo 3 owners, is the model really displayed as "Pico Neo 3" in the settings? The "Pico Neo3 Link" is missing the space before the number. I also think a fallback should be an already existing model and not a newly added one.

Comment (HoLo85) [reply alvr/client_core/src/platform/mod.rs:75]: PLYSHKA responded to this question in Discord. The Neo 3 really has a space in front of the number in the model name.

File: alvr/client_core/src/connection.rs (hunk 1 above)
Comment (HoLo85) [reply alvr/client_core/src/connection.rs]: Reverted the change

**Comment by The-personified-devil** [comment]
You'll need to autoformat the changed code to fix the check

**Comment by HoLo85** [comment]
> You'll need to autoformat the changed code to fix the check

Are those suggestions by rustfmt and are all listed changes marked as error? I think my changes are already formatted for a good readability.

**Comment by The-personified-devil** [comment]
You should format everything according to rustfmt

**Comment by Vixea** [comment]
You need to run `cargo fmt` we don't care if you think you're code is readable to you. You must follow rustfmt's rules.

**Review by The-personified-devil — APPROVED** [review]

**Review by Meister1593 — APPROVED** [review]

**Review by Vixea — APPROVED** [review]
